docs/basics/react-preact.md

```diff
@@ -0,0 +1,364 @@
+
+### Sending a mutation
+
+Let's again pick up an example with an imaginary GraphQL API for todo items, and dive into an
+example! We'll set up a mutation that _updates_ a todo item's title.
+
+```jsx
+const UpdateTodo = `
+  mutation ($id: ID!, $title: String!) {
+    updateTodo (id: $id, title: $title) {
+      id
+      title
+    }
+  }
+`;
+
+const Todo = ({ id, title }) => {
+  const [updateTodoResult, updateTodo] = useMutation(UpdateTodo);
+};
+```
+
+Similar to the `useQuery` output, `useMutation` returns a tuple. The first item in the tuple again
+contains `fetching`, `error`, and `data` — it's identical since this is a common pattern of how
+`urql` presents [operation results](../api/core.md#operationresult).
+
+Unlike the `useQuery` hook, the `useMutation` hook doesn't execute automatically. At this point in
+our example, no mutation will be performed. To execute our mutation we instead have to call the
+execute function — `updateTodo` in our example — which is the second item in the tuple.
+
+### Using the mutation result
+
+When calling our `updateTodo` function we have two ways of getting to the result as it comes back
+from our API. We can either use the first value of the returned tuple — our `updateTodoResult` — or
+we can use the promise that `updateTodo` returns.
+
+```jsx
+const Todo = ({ id, title }) => {
+  const [updateTodoResult, updateTodo] = useMutation(UpdateTodo);
+
+  const submit = newTitle => {
+    const variables = { id, title: newTitle || '' };
+    updateTodo(variables).then(result => {
+      // The result is almost identical to `updateTodoResult` with the exception
+      // of `result.fetching` not being set.
+      // It is an OperationResult.
+    });
+  };
+};
+```
+
+The result is useful when your UI has to display progress on the mutation, and the returned
+promise is particularly useful when you're adding side-effects that run after the mutation has
+completed.
+
+### Handling mutation errors
+
+It's worth noting that the promise we receive when calling the execute function will never
+reject. Instead it will always return a promise that resolves to a result.
+
+If you're checking for errors, you should use `result.error` instead, which will be set
+to a `CombinedError` when any kind of errors occurred while executing your mutation.
+[Read more about errors on our "Errors" page.](./errors.md)
+
+```jsx
+const Todo = ({ id, title }) => {
+  const [updateTodoResult, updateTodo] = useMutation(UpdateTodo);
+
+  const submit = newTitle => {
+    const variables = { id, title: newTitle || '' };
+    updateTodo(variables).then(result => {
+      if (result.error) {
+        console.error('Oh no!', result.error);
+      }
+    });
+  };
+};
+```
+
+There are some more tricks we can use with `useMutation`.<br />
+[Read more about its API in the API docs for it.](../api/urql.md#usemutation)
+
+[On the following page we'll learn about "Document Caching", `urql`'s default caching
```

Comment:
I'd love to replace the final link on each page with "Learning More" or "Reading on" as an H2 heading. This section can contain a list of _all_ things that a person may want to check after, like:

- Learn how to pass fetch options and other things (could be in the same thing)
- Learn how to set up subscriptions
- Normalised caching & Offline-first apps
- Authentication